### PR TITLE
Disable confirmation dialog on localhost

### PR DESCRIPTION
**Pull request by rubenwardy**

### CODE REVIEW

File: src/network/address.cpp [hunk 1]
```diff
@@ -271,3 +271,13 @@ void Address::print(std::ostream *s) const
...
 		*s << serializeString() << ":" << m_port;
 }
+
+bool Address::isLocalhost() const {
+	if (isIPv6()) {
+		static const unsigned char localhost_bytes[] =
+				{ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1 };
+
+		return memcmp(m_address.ipv6.sin6_addr.s6_addr, localhost_bytes, 16) == 0;
+	} else
```
Comment (SmallJoker) [inline src/network/address.cpp]: Needs either brackets for the code block below (style guidelines) or remove `else` (returns anyway).

**Comment by rubenwardy** [comment]
I haven't tested the ipv6 branch, but I copied the code from stackoverflow so it must be correct

File: src/network/address.cpp [hunk 2]
```diff
@@ -271,3 +271,17 @@ void Address::print(std::ostream *s) const
 	else
 		*s << serializeString() << ":" << m_port;
 }
+
+bool Address::isLocalhost() const {
+	if (isIPv6()) {
+		static const unsigned char localhost_bytes[] =
```
Comment (SmallJoker) [inline src/network/address.cpp]: LINT is not happy with it. `// clang-format off`

Comment (paramat) [reply src/network/address.cpp]: Please don't, as we'll end up with thousands of these on/off lines, please whitelist the file instead.

**Comment by paramat** [comment]
I don't consider unit tests necessary for 5.0.0.

**Comment by nerzhul** [comment]
unittests are necessary as a testing part of this new function addition. It's not related to a roadmap but the code provided. As i said some localhost IP are not covered. Unittests will show it just works as intended if the test cases are written.

File: src/network/address.cpp (hunk 2 above)
Comment (nerzhul) [reply src/network/address.cpp]: don't whitelist this pretty file.

Comment (nerzhul) [reply src/network/address.cpp]: did you see that:

```
-bool Address::isLocalhost() const {
+bool Address::isLocalhost() const
+{
```
LINT is okay for this case, and yes it prevent a wrong codestyle here :)
The problem is with the memcmp line

Comment (paramat) [reply src/network/address.cpp]: Yes changes suggested by lint are fine. Whitelisting or clang on/off is no longer needed.

**Comment by nerzhul** [comment]
Fixed by 7796a3118d7b4f58752fad0ca5f676dcafd7a76c & ff5d4ffe1c4b379b0920bef41cdbaebb751f25aa

**Pull request closed without merging**